geonode/facets/apps.py

```diff
@@ -0,0 +1,54 @@
+#########################################################################
+#
+# Copyright (C) 2023 OSGeo
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#########################################################################
+import logging
+
+from django.apps import AppConfig
+
+from geonode.facets.models import FacetProvider
+
+
+logger = logging.getLogger(__name__)
+
+registered_facets = dict()
+
+
+class GeoNodeFacetsConfig(AppConfig):
+    name = "geonode.facets"
+    verbose_name = "GeoNode Facets endpoints"
+
+    def ready(self):
+        from geonode.urls import urlpatterns
+        from . import urls
+
+        urlpatterns += urls.urlpatterns
+
+        init_thesaurus_providers()
```

Comment:
As we did with the geonode-importer it could be good to have a registry for the facets too ([example](https://github.com/GeoNode/geonode-importer/blob/531371daa890421f775bd5f7e9915db42c0b5336/importer/handlers/apps.py#L23-L25))

Reply:
Not possible, since some Facets (for instance the Thesauri) are dynamic and are configured by DB entries. We would need a factory instead.